jvm/sargon-android/build.gradle.kts

```diff
@@ -175,42 +224,101 @@ tasks.register("prepareLocalProperties") {
     tasks.getByName("preBuild").dependsOn(this)
 }
 
-android.libraryVariants.all {
-    val buildType = name
-    val buildTypeUpper = buildType.capitalized()
-
-    val generateBindings = tasks.register(
-        "generate${buildTypeUpper}UniFFIBindings",
-        Exec::class
-    ) {
-        val regenerate = properties["regenerate"] ?: true //TODO improve by using gradle outputs
+afterEvaluate {
```

Comment:
This block sums up the order of tasks that need to run on each occasion. Tasks that involve unit tests don't need to build cargo with ndk, but need to generate bindings. On the other hand tasks that involve building for android or building the integration tests that should run on an android device need to compile ndk binaries and exclude the desktop ones.